### PR TITLE
Introducing black.list

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

## 10

---

- Add new file `/etc/pihole/black.list` that holds the user's blacklist in HOSTS format.
- Use this new file as additional `hosts`.like file in `dnsmasq`'s config
- Add "pihole -g -b" to *only* update black.list (saves a bunch of time when adding/changing only blacklisted files - won't download and subsequentially process all lists, but only processes the blacklist and restarts `dnsmasq`)

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: gravity.sh [hunk 1]
```diff
@@ -299,44 +298,61 @@ gravity_unique() {
...
+}
+
+gravity_hostFormatBlack() {
+  if [[ -f "${blacklistFile}" ]]; then
+    numBlacklisted=$(wc -l < "${blacklistFile}")
+    # Format domain list as "192.168.x.x domain.com"
+    gravity_doHostFormat "${blacklistFile}" "${blackList}.tmp"
+    # Copy the file over as /etc/pihole/black.list so dnsmasq can use it
+    cp "${blackList}.tmp" "${blackList}"
+    rm "${blackList}.tmp"
```
Comment (Mcat12) [inline gravity.sh]: This can be a `mv` statement.

File: gravity.sh [hunk 2]
```diff
@@ -299,44 +298,61 @@ gravity_unique() {
...
+
+gravity_hostFormatGravity() {
+	# Format domain list as "192.168.x.x domain.com"
+	echo "" > "${piholeDir}/${accretionDisc}"
+	gravity_doHostFormat "${piholeDir}/${eventHorizon}" "${piholeDir}/${accretionDisc}"
 	# Copy the file over as /etc/pihole/gravity.list so dnsmasq can use it
-	cp ${piholeDir}/${accretionDisc} ${adList}
-	echo " done!"
+	cp "${piholeDir}/${accretionDisc}" "${adList}"
+	rm "${piholeDir}/${accretionDisc}"
```
Comment (Mcat12) [inline gravity.sh]: .This can be a `mv` statement.

**Comment by p1r473** [comment]
@Mcat12 Hi, if we want to change a blacklist item manually, do we change it in .list or .txt?

**Comment by Mcat12** [comment]
Do not manually modify any files. Use the `pihole -b` commands or the web interface. If you want to discuss this more, open a new topic on discourse.pi-hole.net.

**Comment by PromoFaux** [comment]
This will be especially true of the forthcoming backend changes in the 5.0 release (currently in `development` branch), which will store the user lists in a database, rather than flat files.

**Pull request merged**